components/brave_wallet/browser/brave_wallet_utils_unittest.cc

```diff
@@ -1036,4 +1036,239 @@ TEST(BraveWalletUtilsUnitTest, GetCurrentChainId) {
+      "TnNaU0JqZUQwaU1qVXdJaUJqZVQwaU16VTBJaUJ5UFNJeklpOCtQR05wY21Oc1pTQmplRDBp"
+      "TXpBd0lpQmplVDBpTXpVMElpQnlQU0l6SWk4K1BHTnBjbU5zWlNCamVEMGlNalV3SWlCamVU"
+      "MGlOREUwSWlCeVBTSXpJaTgrUEdOcGNtTnNaU0JqZUQwaU1qQXdJaUJqZVQwaU1qazBJaUJ5"
+      "UFNJeklpOCtQR05wY21Oc1pTQmplRDBpTWpVd0lpQmplVDBpTWprMElpQnlQU0l6SWk4K1BH"
+      "TnBjbU5zWlNCamVEMGlNekF3SWlCamVUMGlNamswSWlCeVBTSXpJaTgrUEM5blBqeHNhVzVs"
+      "WVhKSGNtRmthV1Z1ZENCcFpEMGlZaUlnZURFOUlqRTFNUzR5T0NJZ2VESTlJakUzT0M0eU15"
+      "SWdlVEU5SWpFMU9DNDBOU0lnZVRJOUlqRXdOaTR4SWlCbmNtRmthV1Z1ZEZWdWFYUnpQU0ox"
+      "YzJWeVUzQmhZMlZQYmxWelpTSStQSE4wYjNBZ2IyWm1jMlYwUFNJd0lpQnpkRzl3TFdOdmJH"
+      "OXlQU0lqWm1abUlpOCtQSE4wYjNBZ2IyWm1jMlYwUFNJeElpQnpkRzl3TFdOdmJHOXlQU0lq"
+      "WVRsaE9XRTVJaTgrUEM5c2FXNWxZWEpIY21Ga2FXVnVkRDQ4Y0dGMGFDQm1hV3hzUFNKMWNt"
+      "d29JMklwSWlCa1BTSnRNVGcySURFd09DQXVPU0F5TGpFMkxUZ3VNaUE1TGpnMGFDMHlMalJz"
+      "TFRndU1pMDVMamcwTGprdE1pNHhObWd4TjNwdExqUXhJREk1SUM0ME9TMHhMakUyTFRndU1p"
+      "MDVMamcwYUMweUxqUnNMVGd1TWlBNUxqZzBMalE1SURFdU1UWm9NVGN1T0RKNmJTMHhPUzQ1"
+      "TFRJMUxqWXpMUzQxTVM0eU1uWXlNaTQ0TW13dU5URXVNaklnT0MwNUxqVTVWakV5TVd3dE9D"
+      "MDVMall6ZWsweE5qUWdNVE0wTGpReGRpMHlNaTQ0TW13dExqVXhMUzR5TWkwM0xqazVJRGt1"
+      "TmpOMk5HdzRJRGt1TlRrdU5TMHVNVGg2VFRFMU1TNHpJREV5Tm13dE9DNHlJRGt1T0RRdU5E"
+      "a2dNUzR4Tm1neE55NDRNbXd1TkRrdE1TNHhOaTA0TGpJdE9TNDROR2d0TWk0MGVtMHlPUzR5"
+      "SURJMWRqUnNOQ0EwTGpjemFERXVNRGRoTWpRdU9ERWdNalF1T0RFZ01DQXdJREFnTkM0ME15"
+      "MHhOQzR5TkZZeE5ESnNMVEV1TlRFdExqWXlMVGN1T1RrZ09TNDJNbnB0TFRNeElEQXRPQzA1"
+      "TGpVNExURXVOUzQxT0hZekxqVmhNalF1T0RrZ01qUXVPRGtnTUNBd0lEQWdOQzQwT0NBeE5D"
+      "NHlPR2d4TGpBM2JEUXROQzQzTkMwdU1EVXROQzR3TkhwdE16RXRNekIyTkd3NElEa3VOVGtn"
+      "TVM0MUxTNDFPWFl0TWpKc0xURXVOVEV0TGpZekxUY3VPVGtnT1M0Mk0zcHRNaTQyTkNBME1D"
+      "NHpPRXd4TnpndU55QXhOVFpvTFRJdU5Hd3RPQzR6SURFd2RqTXVObXd4TGpBMkxqWXhZVEkw"
+      "TGprZ01qUXVPU0F3SURBZ01DQXhOQzR3T1MwM0xqUTRiQzB1TURFdE1TNHpOWHBOTVRjNExq"
+      "Y2dNVFV3YkRndU1pMDVMamcwTFM0ME9TMHhMakUyYUMweE55NDRNbXd0TGpRNUlERXVNVFln"
+      "T0M0eUlEa3VPRFJvTWk0MGVrMHhOaklnTVRZNUxqVTRWakUyTm13dE9DNHpMVEV3YUMweUxq"
+      "UnNMVFF1TkRRZ05TNHpOSFl4TGpNMVlUSTFJREkxSURBZ01DQXdJREUwTGpFMElEY3VORGxz"
+      "TVMwdU5ucE5NVGMwTGpVZ01UVXhiQzA0TFRrdU5Ua3RMalV4TGpJeVZqRTJOR3d1TnpNdU16"
+      "WWdOeTQzT0MwNUxqTTJkaTAwZWsweE5qUWdNVFkwZGkweU1pNDBNV3d0TGpVeExTNHlNaTAz"
+      "TGprNUlEa3VOak4yTkd3M0xqYzNJRGt1TXpJdU56TXRMak15ZW0wdE1qSXVORGt0TWprdU16"
+      "Y2dPQzA1TGpVNVZqRXlNV3d0T0MwNUxqVTVMVEV1TlRFdU5UbDJNakpzTVM0MU1TNDJNM3BO"
+      "TVRVekxqY2dNVFV3YkRndU1pMDVMamcwTFM0ME9TMHhMakUyYUMweE55NDRNbXd0TGpRNUlE"
+      "RXVNVFlnT0M0eUlEa3VPRFJvTWk0MGVtMHRPUzQzTFRReUxTNDVJREl1TVRZZ09DNHlJRGt1"
+      "T0RSb01pNDBiRGd1TWkwNUxqZzBMUzQ1TFRJdU1UWm9MVEUzZWlJdlBqd3ZjM1puUGc9PSIs"
+      "ICJhdHRyaWJ1dGVzIjogW3sidHJhaXRfdHlwZSI6ICJTdGF0dXMiLCAidmFsdWUiOiJVbmJ1"
+      "aWx0In1dfQ==";
+  EXPECT_TRUE(ParseDataURIAndExtractJSON(GURL(url), &json));
+  EXPECT_EQ(json, expected);
+}
```

Comment:
I pulled these test vectors from mainnet, so they are a bit large. I'll create some smaller ones to remove the bloat in the test.